_posts/2020-12-11-coordinate-systems.md

```diff
@@ -0,0 +1,18 @@
+---
+layout: post
+title:  "Support for alternate input and output coordinate systems"
+date:   2020-12-11 17:00:00
+author: Rich Gwozdz
+---
+
+With the recent release of Winnow 2.2.0, the Koop Geoservices (FeatureServer) output-plugin now supports alternate coordinate reference systems (CRSs) for input and output data. Specifically:
+
+### Provider's can supply the Geoservices output-plugin with non-WGS84 geospatial data
+Prior to this Winnow release, the Geoservices output-plugin expected providers to supply geospatial data using WGS84 coordinates.  Data using non-WGS84 CRSs had to either (1) convert their data in advance to WGS84, or (2) convert their data to WGS84 on the fly in their provider or in a provider `after` function. Converting to WGS84 on the fly introduces inefficiencies if the output CRS is not WGS84 CRS.  In such cases the data has to be reprojected twice; first, in the provider (to WGS84), and second, in the Geoservices output-plugin (to the target CRS).
+
+Winnow is the Koop dependency that the Geoservices output-plugin uses for post-processing provider GeoJSON, including reprojection. It now supports source data GeoJSON in non-WGS84 CRS.  A provider simply has to either (1) set the GeoJSON `crs` property for the data's CRS, or (2) modify the request's `query` property so that it includes an `inputCrs` property. See the ["Does GeoJSON produced by providers need to use the WGS84 CRS?"](../../../../../docs/basics/faqs#does-geojson-produced-by-providers-need-to-use-the-wgs84-crs) in the FAQs for additional details and references to examples.
+
+It's important to reiterate that support for GeoJSON in non-WGS84 CRSs depends on the output-plugin.  While Koop's default output-plugin, Geoservices (FeatureServer) now supports it, other output plugins (e.g., Vector Tiles) still expect geospatial data that uses WGS84.  These output plugins could be updated to use Winnow to reproject data as necessary - PRs are welcome.
```

Comment:
Do you think we can move this functionality in Koop?

Reply:
Maybe but it presents some problems.  Some output plugins will always require a certain CRS (e.g., VectorTiles and WGS84) where as others will require a CRS that depends on an individual request (e.g. Geoservices and `outSR` value).  Probably can do if we carefully think about the output-plugin specification.

Reply:
Makes sense. This indeed depends on the output format and should be implemented in the output plugin.